README.rst

```diff
@@ -1,63 +1,73 @@
-What is py-esp32-ulp?
----------------------
+=====================
+py-esp32-ulp
+=====================
 
-It is an assembler toolchain for the ESP32 ULP (Ultra Low-Power) Co-Processor,
-written in MicroPython.
+.. image:: ../../actions/workflows/run_tests.yaml/badge.svg
+   :height: 20px
+   :target: ../../actions/workflows/run_tests.yaml
+   :alt: Build Status
 
-It is able to translate small, simple assembler language programs to a
-loadable/executable machine code binary, at runtime, on the ESP32
-microcontroller, from projects implemented in MicroPython.
+py-esp32-ulp is an assembler toolchain for the ESP32 ULP (Ultra Low-Power)
+Co-Processor, written in MicroPython.
 
-This is intended as an alternative approach to assembling such programs on a
-development machine using the binutils-esp32ulp toolchain from Espressif.
+It can translate small assembly language programs to a loadable/executable
+ULP machine code binary, directly on the ESP32 microcontroller.
 
+This is intended as an alternative approach to assembling such programs using
+the binutils-esp32ulp toolchain from Espressif on a development machine.
 
-Status
-------
 
-The most commonly used stuff should work. Many ULP code examples found on
-the web will work unmodified. Notably, assembler macros and #include processing
-are not supported.
+Features
+--------
 
-Expressions in assembly source code are supported and get evaluated during
-assembling. Only expressions evaluating to a single integer are supported.
-Constants defined with ``.set`` are supported in expressions.
+The following features are supported:
 
-We have some unit tests and also compatibility tests that compare the output
-whether it is identical with binutils-esp32ulp output.
+* the entire `ESP32 ULP instruction set <https://esp-idf.readthedocs.io/en/latest/api-guides/ulp_instruction_set.html>`_
+* constants defined with ``.set``
+* constants defined with ``#define``
+* expressions in assembly code and constant definitions
+* RTC convenience macros (e.g. WRITE_RTC_REG)
+* many ESP32 ULP code examples found on the web will work unmodified
 
-There is a simple preprocessor that understands just enough to allow assembling
-ULP source files containing convenience macros such as WRITE_RTC_REG. The
-preprocessor and how to use it is documented here:
-`Preprocessor support <docs/preprocess.rst>`_.
 
-The minimum supported version of MicroPython is v1.12. py-esp32-ulp has been
-tested with MicroPython v1.12 and v1.17. It has been tested on real ESP32
-devices with the chip type ESP32D0WDQ6 (revision 1) without SPIRAM. It has
-also been tested on the Unix port.
+Quick start
+-----------
```

Comment:
could be part of the docs (not of README).

on github, it is not a big issue, but for the package description I think it it a bit much. Also, all stuff using markup (like the code block below) might not look that great in the pypi pkg description.

Reply:
The documentation has a more elaborate version of this. Technically it's a duplication (not of the text, but of the information), but I really like a "quick start" in the README - to see how to use the tool right on the "start page" of the Github repo - so I didn't really want to remove it, but also not put the more elaborate version from the docs into the README. 

Do you feel this "quick start" should be removed entirely from the README?

Reply:
ok, keep it as is for now.